internal/foundationdb_status.go

```diff
@@ -95,3 +95,25 @@ func GetMinimumUptimeAndAddressMap(logger logr.Logger, cluster *fdbv1beta2.Found
 
 	return minimumUptime, addressMap, nil
 }
+
+// DoFaultDomainChecksOnStatusOutput does fault domain checks over the given status object.
+func DoFaultDomainChecksOnStatusOutput(status *fdbv1beta2.FoundationDBStatus) bool {
+	// Storage server related check.
+	for _, tracker := range status.Cluster.Data.TeamTrackers {
+		if !tracker.State.Healthy {
+			return false
+		}
+	}
+
+	// Log server related check.
+	for _, log := range status.Cluster.Logs {
```

Comment:
Do you think it make sense to check all logs generations? Or should be the current epoch good enough?

Reply:
I don't have a clear answer right now. I did add "current" to the status structure, and a @todo item about it (so we can revisit this later).